SYCL/ESIMD/Prefix_Local_sum2.cpp

```diff
@@ -122,53 +121,74 @@ int main(int argc, char *argv[]) {
+          property::queue::enable_profiling{});
 
   auto dev = q.get_device();
   std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
   auto ctxt = q.get_context();
 
   // allocate and initialized input
-  pInputs = static_cast<unsigned int *>(
-      malloc_shared(size * TUPLE_SZ * sizeof(unsigned int), dev, ctxt));
+  unsigned int *pInputs = static_cast<unsigned int *>(
+      malloc(size * TUPLE_SZ * sizeof(unsigned int)));
   for (unsigned int i = 0; i < size * TUPLE_SZ; ++i) {
     pInputs[i] = rand() % 128;
   }
 
-  unsigned sum[2];
-  sum[0] = sum[1] = 0;
-  for (int i = 0; i < 256; i++) {
-    sum[0] += pInputs[i * TUPLE_SZ];
-    sum[1] += pInputs[i * TUPLE_SZ + 1];
-  }
+  // allocate device buffer
+  unsigned int *pDeviceOutputs = static_cast<unsigned int *>(
+      malloc_shared(size * TUPLE_SZ * sizeof(unsigned int), dev, ctxt));
 
   // allocate & compute expected result
   unsigned int *pExpectOutputs = static_cast<unsigned int *>(
       malloc(size * TUPLE_SZ * sizeof(unsigned int)));
   compute_local_prefixsum(pInputs, pExpectOutputs, size);
 
+  // Start Timer
+  esimd_test::Timer timer;
+  double start;
+
+  double kernel_times = 0;
+  unsigned num_iters = 10;
+
   try {
-    auto e1 = q.submit([&](handler &cgh) {
-      cgh.parallel_for<class Accum_iterative>(
-          range<2>{size / PREFIX_ENTRIES, 1} * LocalRange,
-          [=](item<2> it) SYCL_ESIMD_KERNEL {
-            cmk_acum_iterative(pInputs, it.get_id(0), 1, PREFIX_ENTRIES);
-          });
-    });
-    e1.wait();
+    for (int iter = 0; iter <= num_iters; ++iter) {
+      memcpy(pDeviceOutputs, pInputs, size * TUPLE_SZ * sizeof(unsigned int));
+      auto e1 = q.submit([&](handler &cgh) {
+        cgh.parallel_for<class Accum_iterative>(
+            range<2>{size / PREFIX_ENTRIES, 1} * LocalRange,
+            [=](item<2> it) SYCL_ESIMD_KERNEL {
+              cmk_acum_iterative(pDeviceOutputs, it.get_id(0), 1,
+                                 PREFIX_ENTRIES);
+            });
+      });
+      e1.wait();
+      double etime = esimd_test::report_time("kernel time", e1, e1);
+      if (iter > 0)
+        kernel_times += etime;
+      else
+        start = timer.Elapsed();
+    }
   } catch (cl::sycl::exception const &e) {
     std::cout << "SYCL exception caught: " << e.what() << '\n';
     free(pInputs, ctxt);
     free(pExpectOutputs);
-    return e.get_cl_code();
+    return 1;
   }
 
-  bool pass = memcmp(pInputs, pExpectOutputs,
+  // End timer.
+  double end = timer.Elapsed();
+
+  esimd_test::display_timing_stats(kernel_times, num_iters,
+                                   (end - start) * 1000);
+
+  bool pass = memcmp(pDeviceOutputs, pExpectOutputs,
                      size * TUPLE_SZ * sizeof(unsigned int)) == 0;
   std::cout << "Prefix " << (pass ? "=> PASSED" : "=> FAILED") << std::endl
             << std::endl;
 
-  free(pInputs, ctxt);
+  free(pDeviceOutputs, ctxt);
```

Comment:
After this change here and in few other places, 'ctxt' may be removed, I think.
```suggestion
  free(pDeviceOutputs, q);
```

Reply:
Done.